configurator/sshkey/sshkey.go

```diff
@@ -0,0 +1,111 @@
+package sshkey
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sys/unix"
+
+	"github.com/percona/pmm-manage/configurator/config"
+)
+
+func Init(c config.PMMConfig) Handler {
+	return Handler{
+		KeyPath:  c.SSHKeyPath,
+		KeyOwner: c.SSHKeyOwner,
+	}
+}
+
+func (c *Handler) RunSSHKeyChecks() {
+	sshKeyUser, err := user.Lookup(c.KeyOwner)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if c.KeyPath == "" {
+		c.KeyPath = path.Join(sshKeyUser.HomeDir, ".ssh/authorized_keys")
+	}
+
+	sshKeyDir := filepath.Dir(c.KeyPath)
+	logger := log.WithField("dir", sshKeyDir)
+	if dir, err := os.Stat(sshKeyDir); err != nil || !dir.IsDir() {
+		if err := os.MkdirAll(sshKeyDir+"/", 0700); err != nil {
+			logger.WithField("error", err).Fatal("Cannot create ssh directory")
+		}
+		uid, _ := strconv.Atoi(sshKeyUser.Uid)
+		gid, _ := strconv.Atoi(sshKeyUser.Gid)
+		if err := os.Chown(sshKeyDir, uid, gid); err != nil {
+			logger.WithField("error", err).Fatal("Cannot change owner of ssh directory")
+		}
+	}
+	if err := unix.Access(sshKeyDir, unix.W_OK); err != nil {
+		logger.WithField("error", err).Fatal("Cannot write to ssh directory")
+	}
+}
+
+func parse(authorizedKey []byte) (*Key, error) {
+	pubKey, comment, _, _, err := ssh.ParseAuthorizedKey(authorizedKey)
+	if err != nil {
+		return nil, err
+	}
+	return &Key{
+		Type:        pubKey.Type(),
+		Comment:     comment,
+		Fingerprint: ssh.FingerprintSHA256(pubKey),
+	}, err
+}
+
+// return values:
+// 1) *Key   - struct with key which is read from c.KeyPath
+// 2) string - human readable message, which can be shown to the end user (can be localized)
+// 3) error  - debug information which is passed for debug needs, shouldn't be shown to user
+func (c *Handler) Read() (*Key, string, error) {
```

Comment:
Please follow golint conventions, start with method name. Also, this list will not look like you think it will – run `godoc -http=127.0.0.1:6060` locally.
```go
// Read returns …, …, and ….
```